language/stdlib/modules/AccountLimits.move

```diff
@@ -165,26 +157,41 @@ module AccountLimits {
         new_max_inflow: u64,
         new_max_outflow: u64,
         new_max_holding_balance: u64,
+        new_time_period: u64,
     ) acquires LimitsDefinition {
-        assert(has_treasury_compliance_role(tc_account), ENOT_TREASURY_COMPLIANCE);
+        assert(Roles::has_treasury_compliance_role(tc_account), ENOT_TREASURY_COMPLIANCE);
         // As we don't have Optionals for txn scripts, in update_account_limit_definition.move
         // we use 0 value to represent a None (ie no update to that variable)
         let limits_def = borrow_global_mut<LimitsDefinition<CoinType>>(limit_address);
         if (new_max_inflow > 0) { limits_def.max_inflow = new_max_inflow };
         if (new_max_outflow > 0) { limits_def.max_outflow = new_max_outflow };
         if (new_max_holding_balance > 0) { limits_def.max_holding = new_max_holding_balance };
+        if (new_time_period > 0) { limits_def.time_period = new_time_period };
     }
 
-    /// Since we don't track balances of accounts before they are limited, once
-    /// they do become limited the approximate balance in `CointType` held by
-    /// the entity across all of its accounts will need to be set by the association.
-    public fun set_current_holdings<CoinType>(
+    /// Update either the `tracked_balance` or `limit_address` fields of the
+    /// `Window<CoinType>` stored under `window_address`.
+    /// * Since we don't track balances of accounts before they are limited, once
+    ///   they do become limited the approximate balance in `CointType` held by
+    ///   the entity across all of its accounts will need to be set by the association.
+    ///   if `aggregate_balance` is set to zero the field is not updated.
+    /// * If we wish to point this window to a new limits definition, we can do
+    ///   so by passing a `new_limit_address` to this function. If the
+    ///   `new_limit_address` is the `CoreAddresses::VM_RESERVED_ADDRESS()`
+    ///   address, then the `limit_address` field of the window is not updated.
+    public fun update_window_info<CoinType>(
         tc_account: &signer,
         window_address: address,
         aggregate_balance: u64,
+        new_limit_address: address,
     ) acquires Window {
-        assert(has_treasury_compliance_role(tc_account), ENOT_TREASURY_COMPLIANCE);
-        borrow_global_mut<Window<CoinType>>(window_address).tracked_balance = aggregate_balance;
+        assert(Roles::has_treasury_compliance_role(tc_account), ENOT_TREASURY_COMPLIANCE);
+        let window = borrow_global_mut<Window<CoinType>>(window_address);
+        if (aggregate_balance != 0)  { window.tracked_balance = aggregate_balance };
+        if (new_limit_address != CoreAddresses::VM_RESERVED_ADDRESS()) {
```

Comment:
I would vote to omit this this check given that (a) it seems to suggest the possibility that 0x0 can hold a resource (hence Frederic's question!), but it's an important Libra framework invariant that it cannot (hopefully the prover can eventually help us with this) and (b) the next line will already fail with a reasonable error message if this doesn't exist.